app/src/main/java/com/concordium/wallet/ui/bakerdelegation/baker/BakerPoolSettingsActivity.kt

```diff
@@ -0,0 +1,257 @@
+            setCommissionRates()
+        }
+    }
+
+    private fun setCommissionRates() {
+        val chainParams = viewModel.bakerDelegationData.chainParameters ?: return
+        binding.apply {
+            setTransactionRates(chainParams.transactionCommissionRange)
+
+            setBakingRates(chainParams.bakingCommissionRange)
+
+            setRewardRates(chainParams.finalizationCommissionRange)
+        }
+    }
+
+    private val RATE_RANGE_FRACTION_MULTIPLIER = 100000
+
+    @SuppressLint("SetTextI18n")
+    private fun ActivityBakerSettingsBinding.setTransactionRates(
+        transactionRange: TransactionCommissionRange
+    ) {
+        if (transactionRange.min != transactionRange.max) {
+            transactionFeeGroup.visibility = View.VISIBLE
+            transactionFeeMin.text = getMinRangeText(transactionRange.min)
+            transactionFeeMax.text = getMaxRangeText(transactionRange.max)
+            transactionFeeValue.addSuffix("%")
+
+            transactionFeeSlider.apply {
+                isEnabled = true
+                max = getSliderRangeValue(transactionRange.max)
+                min = getSliderRangeValue(transactionRange.min)
+                progress = getSliderDefaultProgressValue(transactionRange.max)
+                transactionFeeValue.setText(getPercentageStringFromProgress(progress))
+
+                setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
+                    override fun onProgressChanged(
+                        seekBar: SeekBar?, progress: Int, fromUser: Boolean
+                    ) {
+                        transactionFeeValue.setText(getPercentageStringFromProgress(progress))
+                        if (transactionFeeValue.hasFocus()) {
+                            transactionFeeValue.clearFocus()
+                            hideKeyboard(rootLayout)
+                        }
+                    }
+
+                    override fun onStartTrackingTouch(seekBar: SeekBar?) {
+                    }
+
+                    override fun onStopTrackingTouch(seekBar: SeekBar?) {
+                    }
+                })
+            }
+        } else {
+            transactionFeeGroup.visibility = View.GONE
+            transactionFeeValue.setText(getPercentageString(transactionRange.max))
+        }
+    }
+
+    private fun ActivityBakerSettingsBinding.setBakingRates(bakingRange: BakingCommissionRange) {
+        if (bakingRange.min != bakingRange.max) {
+            bakingGroup.visibility = View.VISIBLE
+            bakingMin.text = getMinRangeText(bakingRange.min)
+            bakingMax.text = getMaxRangeText(bakingRange.max)
+            bakingValue.addSuffix("%")
+
+            bakingSlider.apply {
+                isEnabled = true
+                max = getSliderRangeValue(bakingRange.max)
+                min = getSliderRangeValue(bakingRange.min)
+                progress = getSliderDefaultProgressValue(bakingRange.max)
```

Comment:
I know I said to default to max, :see_no_evil:  but when updating the pool settings, it should default to the current rate, this value should be available in the bakerPoolInfo of the accountInfo, but doesn't seem to be. It might just be missing from the type, but I'll check tomorrow.

Reply:
I tried to fix it :upside_down_face: